qa/workunits/rgw/tempest.sh

```diff
@@ -0,0 +1,229 @@
+                            --tenant-id "${TEMPEST_TENANT_ID}")
+
+  TEMPEST_USER2_ID=$(get_id tools/with_venv.sh keystone user-create \
+                            --name tempest_u2 \
+                            --pass tempest \
+                            --tenant-id "${TEMPEST_TENANT_ID}")
+
+  # roles
+  ADMIN_ROLE_ID=$(get_id tools/with_venv.sh keystone role-create \
+                            --name admin)
+
+  MEMBER_ROLE_ID=$(get_id tools/with_venv.sh keystone role-create \
+                            --name Member)
+
+  # users privileges
+  tools/with_venv.sh keystone user-role-add \
+                            --user-id ${ADMIN_USER_ID} \
+                            --role-id ${ADMIN_ROLE_ID} \
+                            --tenant-id ${ADMIN_TENANT_ID}
+
+  tools/with_venv.sh keystone user-role-add \
+                            --user-id ${TEMPEST_USER1_ID} \
+                            --role-id ${MEMBER_ROLE_ID} \
+                            --tenant-id ${TEMPEST_TENANT_ID}
+
+  tools/with_venv.sh keystone user-role-add \
+                            --user-id ${TEMPEST_USER2_ID} \
+                            --role-id ${MEMBER_ROLE_ID} \
+                            --tenant-id ${TEMPEST_TENANT_ID}
+
+  # keystone service
+  KEYSTONE_SERVICE_ID=$(get_id tools/with_venv.sh keystone service-create \
+                            --name keystone \
+                            --type identity \
+                            --description "Keystone Identity Service")
+
+  tools/with_venv.sh keystone endpoint-create --region RegionOne \
+                            --service-id ${KEYSTONE_SERVICE_ID} \
+                            --publicurl "http://127.0.1:\$(public_port)s/v2.0" \
+                            --adminurl "http://127.0.0.1:\$(admin_port)s/v2.0" \
+                            --internalurl "http://127.0.0.1:\$(public_port)s/v2.0"
+
+  # object store service
+  SWIFT_SERVICE_ID=$(get_id tools/with_venv.sh keystone service-create \
+                            --name swift \
+                            --type "object-store" \
+                            --description "Swift Service")
+
+  tools/with_venv.sh keystone endpoint-create --region RegionOne \
+                            --service-id ${SWIFT_SERVICE_ID} \
+                            --publicurl "${RADOSGW_URL}" \
+                            --adminurl "${RADOSGW_URL}" \
+                            --internalurl "${RADOSGW_URL}"
+}
+
+deploy_keystone()
+{
+  keystone_download
+
+  if [[ -z "${KESTONE_SKIP_VENV}" ]]; then
+    keystone_make_venv
+  fi
+
+  keystone_configure
+  keystone_start
+  keystone_put_authinfo
+}
+
+tempest_download()
+{
+  rm -rf ${TEMPEST_HOME_DIR}
+  mkdir -p ${TEMPEST_HOME_DIR}
+
+  git clone --quiet -- ${TEMPEST_REPO_URL} ${TEMPEST_HOME_DIR}
+  cd ${TEMPEST_HOME_DIR} \
+      && git checkout ${TEMPEST_RELEASE}
+}
+
+tempest_make_venv()
+{
+  cd ${TEMPEST_HOME_DIR}
+
+  python tools/install_venv.py > /dev/null
```

Comment:
```tools/install_venv.py``` has been recently removed from Keystone. ```tox``` should be used instead but it doesn't work for me. Following workaround may be employed:

```
virtualenv env
source env/bin/activate
pip install -e '.[ldap,memcache,mongodb]'
```